tests/cbmc/proofs/s2n_strcpy/s2n_strcpy_harness.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "api/s2n.h"
+#include "utils/s2n_str.h"
+
+#include <assert.h>
+#include <string.h>
+#include <cbmc_proof/cbmc_utils.h>
+#include <cbmc_proof/make_common_datastructures.h>
+#include <cbmc_proof/proof_allocators.h>
+#include <sys/param.h>
+
+void s2n_strcpy_harness() {
+    char *str = ensure_c_str_is_allocated(MAX_STRING_LEN);
+    const uint32_t slen = strlen(str);
+    const uint32_t buflen;
+    const uint32_t index;
+    __CPROVER_assume(buflen < MAX_STRING_LEN);
+    __CPROVER_assume(index < strlen(str));
+    char buf[buflen];
+
+    /* last must be a pointer into buf */
+    const uint32_t last_offset;
+    __CPROVER_assume(last_offset < buflen);
+    char* last = &buf[last_offset];
+    char str_byte = str[index];
+    char *result;
+
+    /* Non-deterministically set initialized (in s2n_mem) to true. */
+    if (nondet_bool()) {
+        s2n_mem_init();
+    }
+
+    /* Non-deterministically set str to NULL. */
+    if(nondet_bool()) {
+        str = NULL;
+    }
+
+
+    result = s2n_strcpy(buf, last, str);
```

Comment:
you could replace this with
```c
result = s2n_strcpy(buf, last, nondet_bool() ? str : NULL);
```

Reply:
I like it! I've now implemented something similar, but with the nondet_bool determined ahead of time so we know if it was passed a null string.